### PR TITLE
Added Driver for Eddystones

### PR DESCRIPTION
**Pull request by pierantoniomerlino**
This PR adds a new Driver that supports Eddystones. Since they are a sort of beacons, only the listen feature is supported. Read and writes operations have no effect. Moreover, the output of the Driver is a semi-colon separated string like this:

URL;url;txPower;rssi;distance

for URL frame type and

UID;namespace;instance;txPower;rssi;distance

for UID frame type. So, the value.type is not used.
Signed-off-by: pierantoniomerlino <pierantonio.merlino@eurotech.com>

### CODE REVIEW

File: kura/org.eclipse.kura.driver.eddystone.provider/src/main/java/org/eclipse/kura/internal/driver/eddystone/EddystoneDriver.java [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+            EddystoneListener sensorListener = iterator.next();
+            if (sensorListener.getListener().equals(listener)) {
+                iterator.remove();
+            }
+        }
+    }
+
+    @Override
+    public void write(final List<ChannelRecord> records) throws ConnectionException {
+        logger.warn("Write operation not supported");
```
Comment (amitjoy) [inline kura/org.eclipse.kura.driver.eddystone.provider/src/main/java/org/eclipse/kura/internal/driver/eddystone/EddystoneDriver.java:233]: According to contract, it should throw `KuraRuntimeException` with `KuraErrorCode#OPERATION_NOT_SUPPORTED`

File: kura/org.eclipse.kura.driver.eddystone.provider/src/main/java/org/eclipse/kura/internal/driver/eddystone/EddystoneDriver.java [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+        @Override
+        public void close() throws Exception {
+            // Not needed
+        }
+
+        @Override
+        public List<ChannelRecord> execute() throws ConnectionException, KuraException {
+            logger.warn("Read operation not supported");
+            return new ArrayList<>();
```
Comment (amitjoy) [inline kura/org.eclipse.kura.driver.eddystone.provider/src/main/java/org/eclipse/kura/internal/driver/eddystone/EddystoneDriver.java]: You can also return `Collections.emptyList()`

File: kura/org.eclipse.kura.driver.eddystone.provider/src/main/java/org/eclipse/kura/internal/driver/eddystone/EddystoneDriver.java [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+        }
+
+        @Override
+        public List<ChannelRecord> getChannelRecords() {
+            return new ArrayList<>();
```
Comment (amitjoy) [inline kura/org.eclipse.kura.driver.eddystone.provider/src/main/java/org/eclipse/kura/internal/driver/eddystone/EddystoneDriver.java]: You can also return `Collections.emptyList()`

File: kura/org.eclipse.kura.driver.eddystone.provider/src/main/java/org/eclipse/kura/internal/driver/eddystone/EddystoneDriver.java (hunk 2 above)
Comment (pierantoniomerlino) [reply kura/org.eclipse.kura.driver.eddystone.provider/src/main/java/org/eclipse/kura/internal/driver/eddystone/EddystoneDriver.java]: 👍

File: kura/org.eclipse.kura.driver.eddystone.provider/src/main/java/org/eclipse/kura/internal/driver/eddystone/EddystoneDriver.java (hunk 3 above)
Comment (pierantoniomerlino) [reply kura/org.eclipse.kura.driver.eddystone.provider/src/main/java/org/eclipse/kura/internal/driver/eddystone/EddystoneDriver.java]: 👍

File: kura/org.eclipse.kura.driver.eddystone.provider/src/main/resources/org.eclipse.kura.driver.eddystone.provider.dpp [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#Deployment Plugin Project File
```
Comment (MMaiero) [inline kura/org.eclipse.kura.driver.eddystone.provider/src/main/resources/org.eclipse.kura.driver.eddystone.provider.dpp]: I don't believe we need the dpp file. Maybe add a git ignore or delete it.

**Comment by MMaiero** [comment]
We don't have tests for this driver?

File: kura/org.eclipse.kura.driver.eddystone.provider/src/main/resources/org.eclipse.kura.driver.eddystone.provider.dpp (hunk 4 above)
Comment (pierantoniomerlino) [reply kura/org.eclipse.kura.driver.eddystone.provider/src/main/resources/org.eclipse.kura.driver.eddystone.provider.dpp]: Mhh... so we should remove also that for the SensorTag driver.

**Comment by pierantoniomerlino** [comment]
@MMaiero No, we don't have tests for this driver.

**Comment by MMaiero** [comment]
We should try to have some black box tests that obviously simulate a device or what a device shares, in order to catch possible issues/ simplify an eventual process of refactoring.

**Pull request merged**